docs/guide/controller-programming.md

```diff
@@ -431,24 +1314,114 @@ In this case the controller should just save the experiment results and quit by
 This will terminate the controller process and freeze the simulation at the current simulation step.
 The physics simulation and every robot involved in the simulation will stop.
 
+%tab-component "language"
+%tab "C"
 ```c
+#include <stdlib.h>
+
+// freeze the whole simulation
+if (finished) {
+  saveExperimentData();
+  exit(0);
+}
+```
+%tab-end
+
+%tab "C++"
+```cpp
+#include <cstdlib>
+
 // freeze the whole simulation
 if (finished) {
   saveExperimentData();
   exit(0);
 }
 ```
+%tab-end
+
+%tab "Python"
+```python
+import sys
+
+# freeze the whole simulation
+if finished:
+  saveExperimentData()
+  sys.exit(0)
+```
+%tab-end
+
+%tab "Java"
+```java
+// freeze the whole simulation
+if (finished) {
+  saveExperimentData();
+  System.exit(0);
+}
+```
+%tab-end
+
+%tab "MATLAB"
+```MATLAB
+% freeze the whole simulation
+if finished
+  saveExperimentData();
+  quit(0);
+end
+```
+%tab-end
+%end
 
 If only one robot controller needs to terminate but the simulation should continue with the other robots, then the terminating robot should call the `wb_robot_cleanup` function right before quitting:
 
+%tab-component "language"
+%tab "C"
 ```c
+#include <stdlib.h>
+
 // terminate only this robot controller
 if (finished) {
   saveExperimentsData();
   wb_robot_cleanup();
   exit(0);
 }
 ```
+%tab-end
+
+%tab "C++"
+```cpp
+#include <cstdlib>
+
+// terminate only this robot controller
+if (finished) {
+  saveExperimentsData();
+  delete robot;
+  exit(0);
+}
+```
+%tab-end
+
+%tab "Python"
+```python
+import sys
+
+# terminate only this robot controller
+if finished:
+  saveExperimentsData()
+  del robot
+  sys.exit(0)
```

Comment:
Is it possible ? If so what about other languages ?

Reply:
It is indeed, but the `del robot` is probably useless as the Python garbage collector will do the job for you 😉

Reply:
About Java, you can probably use something similar with: ``System.exit(0);``.

Reply:
So is this sentence relevant for other languages than C ?

> If only one robot controller needs to terminate but the simulation should continue with the other robots, then the terminating robot should call the wb_robot_cleanup function right before quitting:

Reply:
Not really, as this function is called from the deletion of the robot class.
You can probably have a look at this cross-languages samples:
https://cyberbotics.com/doc/guide/tutorial-4-more-about-controllers?tab-language=c++#the-controller-code